server/cluster/cluster.go

```diff
@@ -0,0 +1,338 @@
+	}
+	c.metaData.clusterTopology = clusterTopology
+
+	if c.metaData.clusterTopology == nil {
+		return ErrClusterTopologyNotFound.WithCausef("cluster:%v", c)
+	}
+	shardIDs := make([]uint32, len(c.metaData.clusterTopology.ShardView))
+	shardMap := make(map[uint32][]*clusterpb.Shard)
+
+	for _, shard := range c.metaData.clusterTopology.ShardView {
+		shardMap[shard.Id] = append(shardMap[shard.Id], shard)
+		shardIDs = append(shardIDs, shard.Id)
+	}
+	c.metaData.shardMap = shardMap
+	c.metaData.shardIDs = shardIDs
+	return nil
+}
+
+func (c *Cluster) loadShardTopologyLocked(ctx context.Context) error {
+	topologies, err := c.storage.ListShardTopologies(ctx, c.clusterID, c.metaData.shardIDs)
+	if err != nil {
+		return errors.Wrap(err, "cluster loadShardTopologyLocked")
+	}
+	shardTopologyMap := make(map[uint32]*clusterpb.ShardTopology, len(c.metaData.shardIDs))
+	for i, topology := range topologies {
+		shardTopologyMap[c.metaData.shardIDs[i]] = topology
+	}
+	c.metaData.shardTopologyMap = shardTopologyMap
+	return nil
+}
+
+func (c *Cluster) loadSchemaLocked(ctx context.Context) error {
+	schemas, err := c.storage.ListSchemas(ctx, c.clusterID)
+	if err != nil {
+		return errors.Wrap(err, "cluster loadSchemaLocked")
+	}
+	schemaMap := make(map[string]*clusterpb.Schema, len(schemas))
+	for _, schema := range schemas {
+		schemaMap[schema.Name] = schema
+	}
+	c.metaData.schemaMap = schemaMap
+	return nil
+}
+
+func (c *Cluster) loadTableLocked(ctx context.Context) error {
+	for _, schema := range c.metaData.schemaMap {
+		tables, err := c.storage.ListTables(ctx, c.clusterID, schema.Id)
+		if err != nil {
+			return errors.Wrap(err, "cluster loadTableLocked")
+		}
+		for _, table := range tables {
+			if t, ok := c.metaData.tableMap[schema.GetName()]; ok {
+				t[table.GetId()] = table
+			} else {
+				c.metaData.tableMap[schema.GetName()] = map[uint64]*clusterpb.Table{table.GetId(): table}
+			}
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) getSchema(schemaName string) (*Schema, bool) {
+	schema, ok := c.schemas[schemaName]
+	return schema, ok
+}
+
+func (c *Cluster) getTable(ctx context.Context, schemaName, tableName string) (*Table, bool, error) {
+	schema, ok := c.schemas[schemaName]
+	if !ok {
+		return nil, false, ErrSchemaNotFound.WithCausef("schemaName", schemaName)
+	}
+	table, exists := schema.getTable(tableName)
+	if exists {
+		return table, true, nil
+	}
+	// find in storage
+	tablePb, exists, err := c.storage.GetTable(ctx, c.clusterID, schema.GetID(), tableName)
+	if err != nil {
+		return nil, false, errors.Wrap(err, "cluster getTable")
+	}
+	if exists {
+		c.Lock()
+		defer c.Unlock()
+		table := c.updateTableCacheLocked(schema, tablePb)
+		return table, true, nil
+	}
+
+	return nil, false, nil
+}
+
+type MetaData struct {
+	cluster          *clusterpb.Cluster
+	clusterTopology  *clusterpb.ClusterTopology
+	shardIDs         []uint32
+	shardMap         map[uint32][]*clusterpb.Shard          // shard_id -> shard
```

Comment:
No need to suffix the field with its type and it will be better if we call it `shards`.